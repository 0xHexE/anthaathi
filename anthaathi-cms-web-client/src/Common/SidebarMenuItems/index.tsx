import * as React from 'react';
import { SidebarMenuIconButton } from '@anthaathi/web-lib';
import { Block } from 'baseui/block';
import {
  Development,
  BlockStorage,
  MediaLibrary,
  Api,
  Http,
  Settings,
} from '@carbon/icons-react';
import { useRouter } from 'found';

export default function SidebarMenuItems() {
  const { router } = useRouter();

  return (
    <>
      <SidebarMenuIconButton
        key="content"
        icon={<BlockStorage />}
        title="Content"
        onClick={() => {
          router.push('/content');
        }}
      />
      <SidebarMenuIconButton
        icon={<Development />}
        title="Entities"
        key="Entities"
        onClick={() => {
          router.push('/entities');
        }}
      />
      <SidebarMenuIconButton
        icon={<MediaLibrary />}
        title="Media"
        key="Media"
        onClick={() => {
          router.push('/media');
        }}
      />
      <SidebarMenuIconButton
        icon={<Api />}
        title="Playground"
        key="Playground"
        onClick={() => {
          router.push('/playground');
        }}
      />
      <Block $style={{ flexGrow: 1 }} key="grow" />
      <SidebarMenuIconButton
        icon={<Http />}
        key="webhook"
        title="Webhook"
        onClick={() => {
          router.push('/config/webhooks');
        }}
      />
      <SidebarMenuIconButton
        icon={<Settings />}
        key="settings"
        title="Settings"
        onClick={() => {
          router.push('/settings');
        }}
      />
    </>
  );
}
