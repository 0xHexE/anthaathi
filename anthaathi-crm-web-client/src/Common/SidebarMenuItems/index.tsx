import * as React from 'react';
import { SidebarMenuIconButton, SidebarMenuDivider } from '@anthaathi/web-lib';
import { Block } from 'baseui/block';
import {
  Dashboard,
  Settings,
  Calendar,
  Group,
  FileStorage,
  Hashtag,
  Add,
} from '@carbon/icons-react';
import { useRouter } from 'found';
import { useIntl } from 'react-intl';
import { useStyletron } from 'baseui';

export default function SidebarMenuItems() {
  const { router } = useRouter();
  const intl = useIntl();

  const [css] = useStyletron();

  return (
    <>
      <SidebarMenuIconButton
        key="add-new-task"
        icon={<Add />}
        title={intl.formatMessage({
          defaultMessage: 'Create Task',
          id: '4+iiEI',
        })}
        onClick={() => {
          router.push('/task/create');
        }}
        className={css({})}
      />

      <SidebarMenuDivider />

      <SidebarMenuIconButton
        key="overview"
        icon={<Dashboard />}
        title={intl.formatMessage({ defaultMessage: 'Overview', id: '9uOFF3' })}
        onClick={() => {
          router.push('/');
        }}
      />

      <SidebarMenuIconButton
        key="calendar"
        icon={<Calendar />}
        title={intl.formatMessage({ defaultMessage: 'Calendar', id: 'Coy6SH' })}
        onClick={() => {
          router.push('/');
        }}
      />

      <SidebarMenuIconButton
        key="team"
        icon={<Group />}
        title={intl.formatMessage({ defaultMessage: 'Group', id: 'HpoDwk' })}
        onClick={() => {
          router.push('/');
        }}
      />

      <SidebarMenuIconButton
        key="files"
        icon={<FileStorage />}
        title={intl.formatMessage({ defaultMessage: 'Files', id: 'm4vqJl' })}
        onClick={() => {
          router.push('/');
        }}
      />

      <SidebarMenuIconButton
        key="tags"
        icon={<Hashtag />}
        title={intl.formatMessage({ defaultMessage: 'Tags', id: '1EYCdR' })}
        onClick={() => {
          router.push('/');
        }}
      />

      <Block $style={{ flexGrow: 1 }} key="grow" />

      <SidebarMenuIconButton
        icon={<Settings />}
        key="settings"
        title="Settings"
        onClick={() => {
          router.push('/settings');
        }}
      />
    </>
  );
}
