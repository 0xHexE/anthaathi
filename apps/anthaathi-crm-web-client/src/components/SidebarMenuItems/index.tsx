import * as React from 'react';
import {SidebarMenuDivider, SidebarMenuIconButton} from '@anthaathi/web-lib';
import {Block} from 'baseui/block';
import {
  Add,
  Calendar,
  Dashboard,
  FileStorage,
  Group,
  Hashtag,
  Settings,
  // @ts-ignore
} from '@carbon/icons-react';
import {useRouter} from 'next/router';
import {useIntl} from 'react-intl';

export default function SidebarMenuItems() {
  const router = useRouter();
  const intl = useIntl();

  return (
    <>
      <SidebarMenuIconButton
        key="add-new-task"
        icon={<Add />}
        title={intl.formatMessage({
          defaultMessage: 'Create Task',
          id: '4+iiEI',
        })}
        onClick={() => {
          router.push('/task/create');
        }}
      />

      <SidebarMenuDivider />

      <SidebarMenuIconButton
        key="overview"
        icon={<Dashboard />}
        title={intl.formatMessage({defaultMessage: 'Overview', id: '9uOFF3'})}
        onClick={() => {
          router.push('/');
        }}
      />

      <SidebarMenuIconButton
        key="calendar"
        icon={<Calendar />}
        title={intl.formatMessage({defaultMessage: 'Calendar', id: 'Coy6SH'})}
        onClick={() => {
          router.push('/');
        }}
      />

      <SidebarMenuIconButton
        key="team"
        icon={<Group />}
        title={intl.formatMessage({defaultMessage: 'Group', id: 'HpoDwk'})}
        onClick={() => {
          router.push('/');
        }}
      />

      <SidebarMenuIconButton
        key="files"
        icon={<FileStorage />}
        title={intl.formatMessage({defaultMessage: 'Files', id: 'm4vqJl'})}
        onClick={() => {
          router.push('/');
        }}
      />

      <SidebarMenuIconButton
        key="tags"
        icon={<Hashtag />}
        title={intl.formatMessage({defaultMessage: 'Tags', id: '1EYCdR'})}
        onClick={() => {
          router.push('/');
        }}
      />

      <Block $style={{flexGrow: 1}} key="grow" />

      <SidebarMenuIconButton
        icon={<Settings />}
        key="settings"
        title="Settings"
        onClick={() => {
          router.push('/settings');
        }}
      />
    </>
  );
}
