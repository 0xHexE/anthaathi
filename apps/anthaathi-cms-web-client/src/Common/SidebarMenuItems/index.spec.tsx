import * as React from 'react';
import renderer from 'react-test-renderer';
import { makeRouteConfig, Route, createBrowserRouter } from 'found';
import SidebarMenuItems from './index';

describe('SidebarMenuItems', () => {
  it('should render it', () => {
    // This is equivalent:
    const routeConfig = makeRouteConfig(
      <Route path="/" Component={SidebarMenuItems} />,
    );

    const BrowserRouter = createBrowserRouter({
      routeConfig,

      renderError: ({ error }) => (
        <div>{error.status === 404 ? 'Not found' : 'Error'}</div>
      ),
    });

    const tree = renderer.create(<BrowserRouter />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
