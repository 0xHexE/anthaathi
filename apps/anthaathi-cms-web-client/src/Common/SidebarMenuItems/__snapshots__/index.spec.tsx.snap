// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarMenuItems should render it 1`] = `null`;
